hibernate-core/src/main/java/org/hibernate/bytecode/internal/bytebuddy/BytecodeProviderImpl.java

```diff
@@ -83,13 +83,22 @@ public ReflectionOptimizer getReflectionOptimizer(
 			// we only provide a fast class instantiator if the class can be instantiated
 			final Constructor<?> constructor = findConstructor( clazz );
 
-			fastClass = byteBuddyState.load( clazz, byteBuddy -> byteBuddy
-					.with( new NamingStrategy.SuffixingRandom( INSTANTIATOR_PROXY_NAMING_SUFFIX,
-							new NamingStrategy.SuffixingRandom.BaseNameResolver.ForFixedValue( clazz.getName() ) ) )
-					.subclass( ReflectionOptimizer.InstantiationOptimizer.class )
-					.method( newInstanceMethodName )
-							.intercept( MethodCall.construct( constructor ) )
-			);
+			if ( Modifier.isPrivate( constructor.getModifiers() ) ) {
```

Comment:
Nice catch! Are you planning to do that "factory method injection" as well?
Alternatively, couldn't the enhancer make the constructor public?